### PR TITLE
[FIX] App not showing proper alert on team leave

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

First of all: correct the warning when the user decides to leave a team and refactored a part of the function that leaves the team to keep the pattern of the project.

## Issue(s)	

Was showing the incorrect alert when left the team.

## How to test or reproduce

- Enter a team and you can't be the only owner
- Go to actions view
- Tap `Leave`
- You will be redirected to `RoomListView` and the alert showing as if you have been removed

## Screenshots
<table>
<thead>
<tr>
<th>Before</th>
<th>Then</th>
</thead>
</tr>
<tbody>
<tr>
<td>

</td>
<td>

</td>
</tr>
</tbody>
</table>

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/i18n/locales/en.json [hunk 1]
```diff
@@ -721,6 +721,7 @@
   "creating_team": "creating team",
   "team-name-already-exists": "A team with that name already exists",
   "Add_Channel_to_Team": "Add Channel to Team",
+  "Left_The_Team_Successfully":"Left the team successfully",
```
Comment (diegolmello) [inline app/i18n/locales/en.json:724]: Lint is failing

File: app/lib/methods/subscriptions/rooms.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { sanitizedRaw } from '@nozbe/watermelondb/RawRecord';
 import { InteractionManager } from 'react-native';
 
+import I18n from 'i18n-js';
```
Comment (diegolmello) [inline app/lib/methods/subscriptions/rooms.js:4]: It's not from here. It's from `app/i18n`

File: app/lib/methods/subscriptions/rooms.js [hunk 3]
```diff
@@ -301,7 +303,10 @@ export default function subscribeRooms() {
 					const roomState = store.getState().room;
 					// Delete and remove events come from this stream
 					// Here we identify which one was triggered
-					if (data.rid === roomState.rid && roomState.isDeleting) {
+					if (sub._raw.team_main && data.rid === roomState.rid && roomState.isDeleting) {
```
Comment (diegolmello) [inline app/lib/methods/subscriptions/rooms.js:306]: Never use `._raw` directly for this.
You could have done it as `sub.teamMain` in this case

File: app/lib/methods/subscriptions/rooms.js [hunk 4]
```diff
@@ -301,7 +303,10 @@ export default function subscribeRooms() {
...
+						store.dispatch(removedRoom());
+						EventEmitter.emit(LISTENER, { message: I18n.t('Left_The_Team_Successfully') });
```
Comment (diegolmello) [inline app/lib/methods/subscriptions/rooms.js:308]: Can you raise this alert here instead? https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/sagas/room.js#L54

File: app/lib/methods/subscriptions/rooms.js (hunk 3 above)
Comment (diegolmello) [inline app/lib/methods/subscriptions/rooms.js:306]: Can you unify these if conditions and call `store.dispatch(removedRoom());` only once?

File: app/sagas/room.js [hunk 5]
```diff
@@ -64,6 +64,23 @@ const handleLeaveRoom = function* handleLeaveRoom({ rid, t }) {
 	}
 };
 
+const handleLeaveTeam = function* handleLeaveTeam({ room, selected }) {
```
Comment (diegolmello) [inline app/sagas/room.js:67]: Can you undo this change to redux-saga?

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/RoomActionsView/index.js [hunk 6]
```diff
@@ -770,7 +745,7 @@ class RoomActionsView extends React.Component {
 					<List.Item
 						title='Leave'
 						onPress={() => this.onPressTouchable({
-							event: room.teamMain ? this.leaveTeam : this.leaveChannel
+							event: room.teamMain ? this.leaveTeamRoom : this.leaveChannel
```
Comment (diegolmello) [inline app/views/RoomActionsView/index.js:748]: `this.leaveTeamRoom` doesn't exist

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/RoomActionsView/index.js (hunk 6 above)
Comment (reinaldonetof) [reply app/views/RoomActionsView/index.js]: Yes, my fault when I fixed the last conflict. Now it's fixed!

**Review by diegolmello — APPROVED** [review]
